packages/foundry/contracts/FYToken.sol

```diff
@@ -86,6 +86,36 @@ contract FYToken is IFYToken, IERC3156FlashLender, AccessControl, ERC20Permit, C
         emit FlashFeeFactorSet(flashFeeFactor_);
     }
 
+    ///@dev Converts the amount of the principal to the underlying
+    function convertToUnderlying(uint256 principalAmount) external override returns (uint256 underlyingAmount) {
+        return _convertToUnderlying(principalAmount);
+    }
+
+    ///@dev Converts the amount of the principal to the underlying
+    ///Before maturity, returns amount as if at maturity.
+    function _convertToUnderlying(uint256 principalAmount) internal returns (uint256 underlyingAmount) {
+        if (chiAtMaturity == CHI_NOT_SET) {
+            return principalAmount;
+        } else {
+            return principalAmount.wmul(_accrual());
+        }
+    }
+
+    ///@dev Converts the amount of the underlying to the principal
+    function convertToPrincipal(uint256 underlyingAmount) external override returns (uint256 principalAmount) {
+        return _convertToPrincipal(underlyingAmount);
+    }
+
+    ///@dev Converts the amount of the underlying to the principal
+    ///Before maturity, returns amount as if at maturity.
+    function _convertToPrincipal(uint256 underlyingAmount) internal returns (uint256 princpalAmount) {
+        if (chiAtMaturity == CHI_NOT_SET) {
+            return underlyingAmount;
+        } else {
+            return underlyingAmount.wdiv(_accrual());
```

Comment:
`wmul` (used in `_convertToUnderlying`) and `wdiv` (used in `_convertToPrincipal`) both round down, meaning that `x != _convertToUnderlying(_convertToPrincipal(x))` and `x != _convertToPrincipal(_convertToUnderlying(x))`. Such double reversions happen in some integrations and could be troublesome.

In this case, we are fortunate because it is clear that if we round down on `_convertToUnderlying` and up on `_convertToPrincipal` we will always be rounding in favor of the contract, and the double reversions will match.

You can find `wdivup` in `utils-v2`.